GET /utxos/:address & /fees endpoints

altafan opened the pull request:
This adds a BBolt database to keep track of utxos and an endpoint to get utxos from an address.
The database is designed with 3 nested buckets:
- Top-level bucket is named with the btc address
- The mid-level bucket is named `spent`|`unspent`
- The low-level bucket is named with the tx hash of the utxo to be saved

The low-level bucket contains key/value pairs that are respectevely:
- the `double_hash(utxoTxHash + string(utxoTxIndex * 100))`
- the stringified JSON object version of the utxo

When calling `send/:address` endpoint the utxo created owned by `address` is saved in the respective `unspent` bucket
When calling `broadcast/:tx` endpoint the tx inputs are moved from each `unspent`to `spent` bucket, the outputs are added to `unspent`.

In this way we do not need to search sequentially for an utxo saved in the db

This also adds an endpoint `/fees` to estimate fees.  
It returns an object with high, medium and low fees per kb all set to the min relay fee that can be retrieved by quering the deamon with the `getnetworkinfo` method.  
Unfortunately the `btcd/rpcclient` package used does not support this method because it implements the original `bitcoin-cli` ones. This is solved by making an HTTP API call using the `net/http` package like:
```
curl \
-X POST \
-H "Contet-Type: application/json" \
-d '{"jsonrpc":"1.0", "id":"1", "method": "getnetworkinfo","params":[]}' \
http://admin1:123@localhost:19001/
```

altafan committed:
add .gitignore

altafan committed:
update readme

altafan committed:
Add database to keep track of utxos and add get utxos endpoint

tiero commented on router/regtest.go:
```diff
@@ -38,14 +46,18 @@ func (r *RegTest) New() error {
+	r.DB = db
 
+	// Since regtest chain should be empty at this point, lets mine some blocks to increase miner balance
 	mine(r, 200)
```
better doing it outside the New function

altafan replied:
consider that this should be done every time the dockerized daemon cluster is stared. I put it here so this is automated instead of doing it manually or calling a function after the regtest client is initialized with `New()`

tiero replied:
That's fine. I mean we neeld to call mine outside at the same level of New (after). So it can be only a constructor function without side effects

altafan committed:
update .gitignore

altafan committed:
Refactor & fixes:
- fix responses encoding
- add endpoint for fee estimation (http api)
- add block_hash to utxo type object
- prevent from storing utxos with value = 0

altafan committed:
mine blocks outside New function

altafan commented:
Pushed requested changes. @tiero
